### ISSUE TITLE
Integrate real crypto for Shelley mainnet

### ISSUE THREAD

**Issue by edsko** [input-output-hk/ouroboros-network#261]
We need to pick values for the type members of https://github.com/input-output-hk/cardano-ledger-specs/blob/9358418eb1a54c0f264d8d255025e0b020390bb4/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/Crypto.hs#L24.

At the moment we have:
```
instance Crypto TPraosStandardCrypto where
  type DSIGN    TPraosStandardCrypto = Ed25519DSIGN
  type KES      TPraosStandardCrypto = Sum6KES Ed25519DSIGN Blake2b_256
  type VRF      TPraosStandardCrypto = PraosVRF
  type HASH     TPraosStandardCrypto = Blake2b_256
  type ADDRHASH TPraosStandardCrypto = Blake2b_224
```

Before we can deploy on mainnet, we should adopt:

* [x] DSIGN = [Ed25519](https://github.com/input-output-hk/ouroboros-network/pull/1984)
* [X] KES = [real KES](https://github.com/input-output-hk/cardano-base/issues/102) **UPDATE:** we're sticking with `Sum6KES` for mainnet.
* [x] VRF = [PraosVRF](https://github.com/input-output-hk/ouroboros-network/pull/2297)
* [x] HASH = [Blake2b_256](https://github.com/input-output-hk/ouroboros-network/pull/1986)
* [x] ADDRHASH = [Blake2b_224](https://github.com/input-output-hk/cardano-ledger-specs/pull/1549)

**Comment by edsko** [comment]
@nc6 do we need to keep this ticket open, or does #982 subsume it?

**Comment by edsko** [comment]
This ticket _does_ still need to stay open; it can only be closed once we have adopted the new crypto libraries that are currently being considered for KES and VRF.

**Comment by edsko** [comment]
@mrBliss says actually all actual crypto code will live ledger side. If that is true, we can close this after all. Let's verify that.

**Comment by mrBliss** [comment]
> @mrBliss says actually all actual crypto code will live ledger side. If that is true, we can close this after all. Let's verify that.

Ignore what that guy said, he doesn't know what he's talking about :slightly_smiling_face: 

There's the `Crypto` class in cardano-ledger-specs:
https://github.com/input-output-hk/cardano-ledger-specs/blob/e09a3efa9f428de474384856c0311088b1503819/shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Shelley/Crypto.hs

This is how we would currently instantiate that class on the consensus side:
https://github.com/input-output-hk/ouroboros-network/pull/1695/files?file-filters%5B%5D=.hs#diff-8962e0a6804a4df8f80d0ef7c7652ab3R53-R67

AFAIK, the final KES (https://jira.iohk.io/browse/CDT-60) and VRF (https://jira.iohk.io/browse/CDT-59) implementations will follow soon, after which we can replace `SimpleKES` and `SimpleVRF` with the real implementations.

**Comment by edsko** [comment]
> Ignore what that guy said, he doesn't know what he's talking about slightly_smiling_face

I did say "_If_ that is true" :roll_eyes:

**Comment by mrBliss** [comment]
The real implementation that still requires testing: https://github.com/mkannwischer/xmssfs

**Comment by kevinhammond** [comment]
I have updated https://github.com/input-output-hk/cardano-base/issues/102

TLDR: we have been offered a KES implementation in C with a Haskell binding written by @vincenthz.  This should eliminate one of the key outstanding issues that stands in the way of deploying the node.
